multihost_test/Bugzillas/test_subid_ranges_configurable.py

```diff
@@ -0,0 +1,294 @@
+""" Automation of Shadow-utils bugs """
+
+import pytest
+import subprocess
+import time
+import os
+from sssd.testlib.common.utils import SSHClient
+
+
+def exceute_cmd(multihost, command):
+    """ execute cmd in client machine """
+    cmd = multihost.client[0].run_command(command)
+    return cmd
+
+
+class TestSubid(object):
+    """
+    This is for Shadow bugs automation
+    """
+    def test_subid_nochange_nsswitch(self, multihost,
+                                     create_backup,
+                                     create_localuser):
+        """
+        :Title: support pluggable data sources for
+        subid ranges configurable via /etc/nsswitch.conf
+        :id: 6ff93666-00c0-11ec-9be4-845cf3eff344
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1859252
+        :steps:
+            1. Define user range in /etc/subuid
+            2. Define group range in /etc/subgid
+            3. Create a process for user
+            4. Check newuidmap command is working
+            with no change in /etc/nsswitch.conf
+            5. Check newgidmap command is working
+            with no change in /etc/nsswitch.conf
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+            4. Should succeed
+            5. Should succeed
+        """
+        multihost.client[0].run_command("echo 'local_anuj:165536:65536'"
+                                        " > /etc/subuid")
+        multihost.client[0].run_command("echo 'local_anuj:165536:65536' "
+                                        "> /etc/subgid")
+        multihost.client[0].run_command("echo 'unshare -U bash' "
+                                        "> /tmp/anuj.sh")
+        user = 'local_anuj'
+        ssh1 = SSHClient(multihost.client[0].sys_hostname,
+                         username=user, password='Secret123')
+        ssh2 = SSHClient(multihost.client[0].sys_hostname,
+                         username=user, password='Secret123')
+        (results1, results2, results3) = ssh1.exec_command('sh /tmp/anuj.sh')
```

Comment:
What is the purpose of `(results1, results2, results3)`?

Reply:
Other wise console will hang and test will no go forward . As in this case we are going to a different user console .  And our test run as user root

Reply:
I don't understand this explanation.

`results1, results2, results3` are stored but not used. What for?

Reply:
this is the main reason of using results1, results2, results3 . There is no way we can switch to different user console in between while test is running . 'unshare -U bash' command will lead to a console where user will be "nobody"

[aborah@fedora shadow-utils]$ unshare -U bash
[nobody@fedora shadow-utils]$  <<<< ----  here our script will hang .

if we point the result to results1, results2, results3 our test will not hang as the specific command result will be redirected to results1, results2, results3 .

You can try running the script without the results1, results2, results3 as see what happens .

Reply:
Why do you need to run `sh unshare -U bash`? Why not just `unshare -U bash`?

Reply:
if we do just unshare -U bash 
[aborah@fedora shadow-utils]$ unshare -U bash
[nobody@fedora shadow-utils]$ <<<< ---- here we will switch to user "nobody" and our test will stop here .

Reply:
Why does it stop? This happens in ssh session.

Reply:
console will move to nobody@fedora or we will switch to user "nobody" . I can explain to you with live demo . Lest me know if you have time now

Reply:
I don't want a demo, I want an explanation :)

Reply:
if you do unshare -U bash .

[aborah@fedora shadow-utils]$ unshare -U bash
[nobody@fedora shadow-utils]$ <<<<---  Here you see we logged in as user "nobody". (nobody@fedora) How the script will keep running if we switch user in between the script ? This is the only explanation i have . I am open for live demo if you want

Reply:
But this happens inside ssh session. This doesn't affect the user python script is run under!

I'm not so sure about `multihost.client[0].run_command(command)` - I don't know how is it implemented. But changing user in ssh session (actually running new shell under another user) shouldn't cause any issues.